godel_plugins/src/widgets/robot_blending_widget.cpp

```diff
@@ -429,244 +491,126 @@ void RobotBlendingWidget::save_robot_scan_parameters()
-	surface_detection_parameters_.rg_neightbors = ui_.LineEditRgNeighbors->text().toDouble();
-	surface_detection_parameters_.rg_smoothness_threshold = DEG2RAD(ui_.LineEditRgSmoothnessThreshold->text().toDouble());
-	surface_detection_parameters_.rg_curvature_threshold = ui_.LineEditRgCurvatureThreshold->text().toDouble();
-
-	surface_detection_parameters_.voxel_leafsize = ui_.LineEditVoxelLeaf->text().toDouble();
-	surface_detection_parameters_.tabletop_seg_distance_threshold = ui_.LineEditTabletopSegmentationDist->text().toDouble();
-	surface_detection_parameters_.use_tabletop_seg = ui_.CheckBoxUseTabletopSegmentation->isChecked();
-	surface_detection_parameters_.ignore_largest_cluster = ui_.CheckBoxIgnoreLargestCluster->isChecked();
-
-	surface_detection_parameters_.mls_point_density = ui_.LineEditMlsPointDensity->text().toDouble();
-	surface_detection_parameters_.mls_upsampling_radius = ui_.LineEditMlsUpsamplingRadius->text().toDouble();
-	surface_detection_parameters_.mls_search_radius = ui_.LineEditMlsSearchRadius->text().toDouble();
-
-	surface_detection_parameters_.tr_search_radius = ui_.LineEditTrSearchRadius->text().toDouble();
-	surface_detection_parameters_.tr_mu = ui_.LineEditTrMu->text().toDouble();
-	surface_detection_parameters_.tr_max_nearest_neighbors = ui_.LineEditTrNearestNeighbors->text().toDouble();
-	surface_detection_parameters_.tr_max_surface_angle = DEG2RAD( ui_.LineEditTrMaxSurfaceAngle->text().toDouble());
-	surface_detection_parameters_.tr_min_angle = DEG2RAD(ui_.LineEditTrMinAngle->text().toDouble());
-	surface_detection_parameters_.tr_max_angle = DEG2RAD(ui_.LineEditTrMaxAngle->text().toDouble());
-	surface_detection_parameters_.tr_normal_consistency = ui_.CheckBoxTrNormalConsistency->isChecked();
-
-	surface_detection_parameters_.pa_enabled = ui_.CheckBoxPaEnabled->isChecked();
-	surface_detection_parameters_.pa_seg_max_iterations = ui_.LineEditPaSegMaxIterations->text().toInt();
-	surface_detection_parameters_.pa_seg_dist_threshold = ui_.LineEditPaSegDistThreshold->text().toDouble();
-	surface_detection_parameters_.pa_sac_plane_distance = ui_.LineEditPaSACPlaneDistance->text().toDouble();
-	surface_detection_parameters_.pa_kdtree_radius = ui_.LineEditPaKdtreeRadius->text().toDouble();
-}
+	if (!item) return;
 
-void SurfaceDetectionConfigWidget::accept_changes_handler()
-{
-	save_parameters();
-	Q_EMIT parameters_changed();
-	hide();
-}
+	QString old_text = item->text();
+	// spawn window to prompt a rename
+	QString new_text = QInputDialog::getText(this, "Surface Rename", "Enter a new surface name: ");
 
-void SurfaceDetectionConfigWidget::cancel_changes_handler()
-{
-	hide();
+	if (!new_text.isEmpty())
+	{
+		godel_msgs::RenameSurfaceResponse res;
+		godel_msgs::RenameSurfaceRequest req;
+		req.old_name = old_text.toStdString();
+		req.new_name = new_text.toStdString();
+		if (rename_surface_client_.call(req, res))
+		{
+			item->setText(new_text);
```

Comment:
Yes. For the sake of these pull requests, I have:
 * Made all new files follow ROS conventions (2-wide spaces)
 * Kept original files tabbed.

Because most of the original Godel code was tab spaced, changing it all in the pull requests would read as an absolute sea of changes. It's very easy to change with editors, though, so it's a prime candidate for an issue/immediate pull request after.